### PR TITLE
Remove inline css dcr ad slot fluid

### PR DESCRIPTION
**Pull request by liywjl**
## What does this change?
Remove inline CSS for DCR, and move CSS logic to DCR

## Does this change need to be reproduced in dotcom-rendering ?

- [ ] No
- [X] Yes
Need to make sure https://github.com/guardian/dotcom-rendering/pull/2017 is merged before merging this PR

### Images from DCR

| Before      | After      |
|-------------|------------|
| ![before][] | ![after][] |

[before]: https://user-images.githubusercontent.com/8831403/97888730-645fe880-1d23-11eb-860d-514c35b4154b.png
[after]: https://user-images.githubusercontent.com/8831403/97888827-822d4d80-1d23-11eb-8517-6b12e4f8ffbc.png

refer to https://www.theguardian.com/us-news/2020/oct/29/lordstown-ohio-trump-gm-plant-election

## Checklist

- Make sure https://github.com/guardian/dotcom-rendering/pull/2017 is merged before merging this PR

### Does this affect other platforms?

- [ ] AMP 
- [ ] Apps
- [ ] Other (please specify)

### Does this affect GLabs Paid Content Pages? Should it have support for Paid Content?

- [x] No
- [ ] Yes (please give details)

### Does this change break ad-free?

- [x] No
- [ ] It did, but tests caught it and I fixed it
- [ ] It did, but there was no test coverage so I added that then fixed it

### Does this change update the version of CAPI we're using?

- [x] No, all the existing database files are just fine
- [ ] Yes, and I have [re-run all the tests locally and checked in all the updated data/database/xyz files](https://github.com/guardian/frontend/blob/main/docs/03-dev-howtos/15-updating-test-database.md)

### Accessibility test checklist

- [ ] [Tested with screen reader](https://accessibility.gutools.co.uk/testing/web/screen-readers/)
- [ ] [Navigable with keyboard](https://accessibility.gutools.co.uk/testing/web/keyboard-navigation/)
- [ ] [Colour contrast passed](https://accessibility.gutools.co.uk/testing/web/colour-contrast/)

### Tested

- [ ] Locally
- [ ] On CODE (optional)

### CODE REVIEW

File: static/src/javascripts/projects/commercial/modules/dfp/render-advert.js [hunk 1]
```diff
@@ -33,23 +32,6 @@ const addClassIfHasClass = (newClassNames: Array<string>) =>
                     newClassNames.forEach(className => {
                         advert.node.classList.add(className);
                     });
-                    // Add fluid styles from _adslot.scss
```
Comment (ioanna0) [inline static/src/javascripts/projects/commercial/modules/dfp/render-advert.js:36]: Seems like there was a bug in there, good catch! 👍 (this block was there to mirror the `.ad-slot--fluid` css rules into DCR.)

 I think it comes down to the fact that `newClassNames` don't have the `ad-slot--im` class in it when called in [here](https://github.com/guardian/frontend/blob/f3a6d6a7894703798ab7d3e885aa665f955f15d7/static/src/javascripts/projects/commercial/modules/dfp/render-advert.js#L60) 

Could we then make sure [these rules](https://github.com/guardian/frontend/blob/ba6ff947c2551f98da59dc7252dda78bc95c6a4e/static/src/stylesheets/module/_adslot.scss#L413) are properly migrated into DCR, potentially with some global styles?

**Review by ioanna0 — APPROVED** [review]

**Review by mxdvl — APPROVED** [review]
👍 LGTM

PS. I took the liberty to update the table markdown formatting to get a side-by-side screenshot comparison!

File: static/src/javascripts/projects/commercial/modules/dfp/render-advert.js (hunk 1 above)
Comment (mxdvl) [reply static/src/javascripts/projects/commercial/modules/dfp/render-advert.js:36]: I feel like some of the styles are missing from https://github.com/guardian/dotcom-rendering/pull/2017

**Review by mxdvl — CHANGES_REQUESTED** [review]
After some further consideration, I fear that introduces a regression in the fluid ad slots, which were initially fixed in #22751.

Have you found no issues with fluid ads when running locally or on CODE?

**Review by shtukas — APPROVED** [review]

**Comment by liywjl** [comment]
> After some further consideration, I fear that introduces a regression in the fluid ad slots, which were initially fixed in #22751.
> 
> Have you found no issues with fluid ads when running locally or on CODE?

I have run everything on CODE, it seems to work fine. Note that in order to test this PR you need to deploy both this PR and https://github.com/guardian/dotcom-rendering/pull/2017/files to CODE. When running in parallel you can see that the styles are applied correctly. 

Regarding https://github.com/guardian/frontend/pull/22751, I think it would make sense to migrate all of this logic over to DCR instead of having it added using the `isDotcomRendering`. I will create a ticket for this.

**Comment by mxdvl** [comment]
[Here is a table][ads] showing all the different styles that need to have parity between DCR and `frontend`. 

[ads]: https://docs.google.com/spreadsheets/d/1gq5ODcdDHV_bQzM_UGipm-3N8iDF_j3hba1W4QfiGTg/edit#gid=0 "DCR - Native format testing"

**Comment by liywjl** [comment]
@mxdvl @ioanna0 I have removed some more CSS styles based on the PR you linked here: https://github.com/guardian/frontend/pull/22751 

Since that PR has been merged, there have been more conditional DCR CSS added. Because styles are conditional, I am unsure how easy it is to translate them to DCR without better understanding those conditions. Here are 2 notable ones from `background.js`:

https://github.com/guardian/frontend/blob/main/static/src/javascripts/projects/commercial/modules/messenger/background.js#L123

I understand that this is to apply styles to the 1st ad block, but trying to determine what context that ad slot is, is quite challenging. Also it appears that this CSS styles `contain: layout` is unique to DCR? was this placed here in order to fix something on DCR?

https://github.com/guardian/frontend/blob/main/static/src/javascripts/projects/commercial/modules/messenger/background.js#L145

This is only added if the condition `specs.scrollType === 'interscroller'` is true (am unsure what the means, and is it something in DCR we could detect on our end?). 
Why does DCR need `bottomLine` DOM element? Would this not create a difference between DCR and frontend?
I could not find anything with a className of `ad-slot__line` both in frontend and DCR, so is there a reason for this? 


As mentioned in the commercial-DCR catchup, there are actually a lot of styles from `_adslot.scss` and `_creatives.scss` that are not present in DCR. Therefore it might be interesting to explore the idea of encapsulating this logic within the rendering of these DOM elements, rather than have duplicate copies of the CSS in DCR and front maybe  Maybe it is better to explore the idea of encapsulating these styles.

**Comment by liywjl** [comment]
After several discussions on what would be the best approach, I think there is some agreement on moving the SCSS that is related to commercial to be bundled separately from frontend's SCSS. We could even explore dynamically importing some of the lesser used SCSS as a way of reducing the bundle size using webpack. Feel free to reach out if you are interested in pairing on this.

**Pull request closed without merging**